application/classes/Ushahidi/Repository/Form/Stats.php

```diff
@@ -87,6 +88,80 @@ public function getResponses($form_id)
 			->execute($this->db)
 			->get('total');
 	}
+
+	public function countTotalPending($form_id, $total_sent) {
+		$form_id = intval($form_id);
+		//total_contacts = select count(contact_id) from targeted_survey_state where form_id=8 and survey_status IN ('RECEIVED RESPONSE','PENDING RESPONSE')
+		//total_attributes = select count(form_attributes.id) from form_attributes INNER JOIN form_stages ON form_attributes.form_stage_id = form_stages.id WHERE form_stages.form_id=8
+		$total_contacts = DB::query(Database::SELECT,
+			"select count(contact_id) as total from targeted_survey_state where form_id=$form_id
+			 and survey_status NOT IN ('SURVEY FINISHED')")
+		->execute($this->db)->get('total');
+		$total_attributes = DB::query(Database::SELECT,
+			"select count(form_attributes.id) as total from form_attributes 
+			INNER JOIN form_stages ON form_attributes.form_stage_id = form_stages.id WHERE form_stages.form_id=$form_id")
```

Comment:
I can't seem to find docs for this now, but is this the right/safe way to bind params in a query?

Reply:
as mentioned in the TODO list, I need to move this to the query builder. So no, it's not

Reply:
just in case:the rough query (replacing the form id obviously)  is this
```SELECT SUM(results.counted) as total FROM (SELECT count(form_attributes.form_stage_id) as counted from targeted_survey_state INNER JOIN form_stages ON targeted_survey_state.form_id=form_stages.form_id INNER JOIN form_attributes ON form_stages.id = form_attributes.form_stage_id INNER JOIN (SELECT form_attributes.priority, targeted_survey_state.form_attribute_id, targeted_survey_state.contact_id FROM form_attributes INNER JOIN targeted_survey_state ON form_attributes.id =targeted_survey_state.form_attribute_id WHERE targeted_survey_state.form_id=8 and targeted_survey_state.survey_status='INACTIVE') as internal_query ON internal_query.contact_id = targeted_survey_state.contact_id WHERE form_attributes.priority > internal_query.priority AND survey_status = 'INACTIVE' AND form_stages.form_id=8 group by targeted_survey_state.contact_id, form_attributes.form_stage_id) as results; ```